install/first_google_boot.sh

```diff
@@ -222,17 +222,38 @@ function extract_spinnaker_kube_credentials() {
     gcloud container clusters get-credentials $kube_cluster --zone $kube_zone --project $kube_project
 
     if [[ -s $config_path ]]; then
-      echo "Kubernetes credentials successfully extracted to $config_path" 
+      echo "Kubernetes credentials successfully extracted to $config_path"
       chmod 400 $config_path
       chown spinnaker:spinnaker $config_path
+
+      write_default_value "SPINNAKER_KUBERNETES_ENABLED" "true"
     else
-      echo "Failed to extract kubernetes credentials to $config_path"  
+      echo "Failed to extract kubernetes credentials to $config_path"
       rm $config_path
     fi
-    write_default_value "SPINNAKER_KUBERNETES_ENABLED" "true"
+
+    # Safely return since we have attempted to configure Kubernetes with the
+    # provided metadata.
+    return
   else
     clear_instance_metadata "kube_cluster"
   fi
+
```

Comment:
Logically, you can't specify both a cluster & a config file since the credentials for one would overwrite the other.